presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SwapAdjacentWindowsByPartitionsOrder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.SymbolAllocator;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.WindowNode;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Optional;
+
+import static com.facebook.presto.sql.planner.iterative.rule.Util.transpose;
+
+public class SwapAdjacentWindowsByPartitionsOrder
+        implements Rule
+{
+    @Override
+    public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator)
+    {
+        if (!(node instanceof WindowNode)) {
+            return Optional.empty();
+        }
+
+        WindowNode parent = (WindowNode) node;
+
+        PlanNode child = lookup.resolve(parent.getSource());
+        if (!(child instanceof WindowNode)) {
+            return Optional.empty();
+        }
+
+        Comparator<WindowNode> comparator = new PartitionByComparator();
+        if (comparator.compare(parent, (WindowNode) child) < 0) {
+            return Optional.of(transpose(parent, child));
```

Comment:
Yes, it is. This is a problem referred here: https://github.com/prestodb/presto/issues/7247
I have a patch for `MergeWindow` old optimizer, here: https://github.com/prestodb/presto/pull/7298.
Addressed this problem here as well (a906f7955487af84e74fc2d43fcf28952e4af2b7).